README
The purpose of Rescapp, is to provide the user with a graphical rescue tool that will 
assist them to regain access to a computer that has become non-bootable.

You will also be able to regain access to a Windows based system that has had the 
password forgotten, thanks to the chntpw package.

It is recommended to use the chntpw package made by rescatux, as this is different
to the package that is provided by your Distro.  It has had features added to
automate the password change process as much as possible.

It is available from:

https://github.com/rescatux/chntpw

Once there, click on the word "releases" which will take you to the downloads
section.

You will be able to either blank the password, and then change it once you reboot
into Windows, or you can opt to change the password through this tool.  However, 
it is recommended that you use a temporary password, and change it once you have 
regained access to the computer, so that you can use the correct encryption 
algorithm provided by the version of Windows that you are using.

Rescapp makes use of a number of programs, such as gparted which is used for
hard drives that have been partitioned as GUID (GPT). This type of partitioning
is required for hard drives that are greater than 2TB, but can also be used
for smaller drives, and is the prefered method for booting systems that also
use UEFI.

You can also use Rescapp, to regain access to systems that use SELinux for
hardend security.  You will be guided through the restoration process, as
Rescapp is a rescue "Wizard", providing inline help, as well as assistance
via chat or via the support forums.

Rescapp, is a collection of bash scripts, and makes use of Python3 and also
QT5.

If you are a Developer, and have a little bit of spare time, you can contribute
to this project either as a one time only contribution, ie you have taken a look
at the "todo" within the code, and find that it is a very quick implimentation,
then you clone the https://github.com/rescatux/rescapp.git repository locally,
create a new branch and then commit your changes and push them to an account
that you have on github, and create a pull request, or ongoing, as time permitts.
For this, you will need to contact Adrian, who is the developer, via the issue tracker, 
on https://github.com/rescatux/rescapp/issues and ask which branch he is currently
using for development, so that you can create the pull request on the correct
branch.

Rescapp, is also available on a live Debian cd/dvd/usb ISO, known as rescatux.

For full documentation and features, please visit the rescatux website:

https://www.supergrubdisk.org/rescatux/

To know how to install Rescapp, please read the INSTALL file in the source code tarball.

To know how to publish Rescapp, please read the PUBLISH file in the source code tarball.

If you happen to find a bug in Rescapp, please use the issue tracker on github:

https://github.com/rescatux/rescapp/issues

If you happen to find a bug in the chntpw package provided by rescatux, please use the
issue tracker on github:

https://github.com/rescatux/chntpw/issues

Please remember to use the correct issue tracker for the package that you have found
an issue with, as this saves time in having to move it to the correct location.




debian/changelog
rescapp (0.57-1) unstable; urgency=low

  * Support chat rooms have been moved to Libera IRC.

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 14 Nov 2021 15:43:00 +0000

rescapp (0.56-1) unstable; urgency=low

  * Rest of the plugins documentation has been updated to match 2019 gui.
  * Removed tooltips from main menu.
  * Added initial useful explanations on options documentation.
  * Disable action buttons when an option is run.
  * UEFI options now show the final user when the backup of EFI directory fails.
  * README reorganisation.
  * Stable version.

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sat, 28 Mar 2020 21:54:00 +0000

rescapp (0.54b6-1) unstable; urgency=low

  * Added PUBLISH instructions.
  * Removed non English translations from internal documentation.
  * winmbr option is now stable.
  * Local documentation build process has been improved to be based on a template.
  * Add experimental logo to Rescapp about page.
  * Added new Rescapp desktop icons and a window icon
  * AFD Scanning technology has been improved with new United States of Spain gimmick
  * Show Rescapp help right at the Rescapp startup.
  * Many plugins documentation have been updated to match 2019 gui

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sat, 18 Jan 2020 21:01:00 +0000

rescapp (0.54b5-1) unstable; urgency=low

  * Handle non found devices properly on the rest of options

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sat, 18 Jan 2020 21:01:00 +0000

rescapp (0.54b4-1) unstable; urgency=low

  * Minor inxi improvements
  * Check for Hexchat binary
  * Handle non found devices properly on many options

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Mon, 02 Dec 2019 21:13:00 +0000

rescapp (0.54b3-1) unstable; urgency=low

  * Gparted, Photorec and Testdisk are now checked before running them
  * Add an specific non-documented documentation page.
  * Added System Info Script (Inxi) option
  * bootinfoscript was updated to use system's installed bootinfoscript

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 17 Nov 2019 17:59:00 +0000

rescapp (0.54b2-1) unstable; urgency=low

  * More compact UI
  * Added rescapp title and version to the main UI
  * Advice to choose an option to read its manual
  * After selecting an option it shows its name in the option screen
  * Advice to read the manual before hitting run on the option screen
  * New status label shows the option being ready to run, running or finished
  * New messages box that show the actions the option is performing

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 27 Oct 2019 10:00:00 +0000

rescapp (0.54b1-1) unstable; urgency=low

  * Do not mount neither root partition nor tmp partition (Improves Ubuntu grub fix)
  * Moved from python2 to python3
  * Removed gddrescue, myrescue and gparted dependencies

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 15 Sep 2019 15:02:00 +0000

rescapp (0.53b3-1) unstable; urgency=low

  * Added Xterm self test
  * Remove back button and history
  * Fixed: Non secure boot UEFI mode makes rescapp to crash before its menu is shown.

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 15 Sep 2019 15:02:00 +0000

rescapp (0.53b2-1) unstable; urgency=low

  * Fixed UEFI Secure Boot self test.

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Tue, 07 May 2019 21:04:00 +0000

rescapp (0.53b1-1) unstable; urgency=low

  * Updated to 0.53b1 upstream version
  * Added Selftest feature.
  * New samunlock self test.
  * New Rescatux Live CD self test.
  * New SELinux self test.
  * New UEFI Boot self test.
  * New UEFI Secure Boot self test.
  * Documented new mokutil package dependency

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sat, 20 Apr 2019 14:32:00 +0000

rescapp (0.52b6-1) unstable; urgency=low

  * Updated to 0.52b6 upstream version
  * Avoid Unknown GNU/Linux distro breaking zenity dialogs.
  * Improved Fedora/CentOS/RHEL distribution name detection
  * Documented new os-prober package dependency

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 31 Mar 2019 15:58:00 +0000

rescapp (0.52b5-1) unstable; urgency=low

  * Updated to 0.52b5 upstream version
  * Iceweasel is not longer a requirement. xdg-utils is.

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 03 Mar 2019 12:03:00 +0000

rescapp (0.52b4-1) unstable; urgency=low

  * Updated to 0.52b4 upstream version
  * chat: Use temp hexchat directory to avoid overwriting personal hexchat directories on non live installations.

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 03 Mar 2019 12:03:00 +0000

rescapp (0.52b3-1) unstable; urgency=low

  * Updated to 0.52b3 upstream version
  * helpers/rescapp-set-hard-disks-boot-order ported to PyQT5
  * lib/rescapp_afd* (English version) ported to PyQT5
  * lib/rescapp_afd* (Spanish version) ported to PyQT5

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 10 Dec 2017 11:50:53 +0000

rescapp (0.52b2-1) unstable; urgency=low

  * Updated to 0.52b2 upstream version
  * Removed SELinux dependencies

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 10 Dec 2017 10:25:14 +0000

rescapp (0.52b1-1) unstable; urgency=low

  * Updated to 0.52b1 upstream version
  * Dependencies no longer based on qt4 but qt5

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sun, 10 Dec 2017 01:06:42 +0000

rescapp (0.51b5-1) unstable; urgency=low

  * Updated to 0.51b5 upstream version

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sat, 09 Dec 2017 16:51:09 +0000

rescapp (0.51b4-4) unstable; urgency=low

  * testdisk dependencies were simplified
  * efibootmgr dependencies were simplified
  * Removed samdump2 and bkhive non used packages
  * Converted SELinux dependencies into Stretch ones

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sat, 09 Dec 2017 15:59:46 +0000

rescapp (0.51b4-1) unstable; urgency=low

  * xchat was replaced by hexchat

 -- Adrian Gibanel Lopez <adrian15sgd@gmail.com>  Sat, 09 Dec 2017 15:56:08 +0000


lib/rescapp_afd2_es.py
# -*- coding: utf-8 -*-
# April Fools Day 2 Script
# Copyright (C) 2012,2013,2014,2015,2016,2017,2018,2019,2020 Adrian Gibanel Lopez
afd2_app = QtWidgets.QApplication(sys.argv)


afd2_grid = QtWidgets.QGridLayout()
afd2_grid.setSpacing(10)

# afd2_title
afd2_title = QtWidgets.QLabel(u"¿Asustado?")
afd2_title_font = QtGui.QFont()
afd2_title_font.setPointSize(30)
afd2_title.setFont(afd2_title_font)

afd2_title_palette = afd2_title.palette()
afd2_title_palette.setColor(afd2_title.backgroundRole(),QtCore.Qt.red)
afd2_title_palette.setColor(afd2_title.foregroundRole(),QtCore.Qt.red)
afd2_title.setPalette(afd2_title_palette)

# afd2_message

this_year = str(datetime.datetime.now().year)

afd2_message = QtWidgets.QLabel(u"¡¡¡ " + u"Feliz día<br>de los<br>santos inocentes<br>del año "+this_year+ u" !!!")
afd2_message_font = QtGui.QFont()
afd2_message_font.setPointSize(40)
afd2_message.setFont(afd2_message_font)

afd2_message_palette = afd2_message.palette()
afd2_message_palette.setColor(afd2_message.backgroundRole(),QtCore.Qt.black)
afd2_message_palette.setColor(afd2_message.foregroundRole(),QtCore.Qt.black)
afd2_message.setPalette(afd2_message_palette)

# afd2_pay

next_year = str(datetime.datetime.now().year + 1)

afd2_pay = QtWidgets.QLabel("Pulsa en cualquier sitio para usar Rescatux / Rescapp .")
afd2_pay_font = QtGui.QFont()
afd2_pay_font.setPointSize(14)
afd2_pay.setFont(afd2_pay_font)

afd2_pay_palette = afd2_pay.palette()
afd2_pay_palette.setColor(afd2_pay.backgroundRole(),QtCore.Qt.darkGreen)
afd2_pay_palette.setColor(afd2_pay.foregroundRole(),QtCore.Qt.darkGreen)
afd2_pay.setPalette(afd2_pay_palette)

afd2_grid.addWidget(afd2_title,0,0,1,0)
afd2_grid.addWidget(afd2_message,5,0,2,0)
afd2_grid.addWidget(QtWidgets.QLabel(),6,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),7,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),8,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),9,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),10,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),11,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),12,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),13,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),14,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),15,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),16,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),17,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),18,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),19,0,2,1)
afd2_grid.addWidget(QtWidgets.QLabel(),20,0,2,1)
afd2_grid.addWidget(afd2_pay,21,0,1,1)

afd2_scrollArea = VerticalScrollArea()
afd2_gridQWidget =  QtWidgets.QWidget()
afd2_gridQWidget.setLayout(afd2_grid)
afd2_scrollArea.setWidgetResizable(False)
afd2_scrollArea.setWidget(afd2_gridQWidget)
afd2_scrollArea.setMinimumWidth(afd2_gridQWidget.minimumSizeHint().width())
afd2_qVboxLayout = QtWidgets.QVBoxLayout()
afd2_qVboxLayout.addWidget(afd2_scrollArea)

afd2_window = CloseOnClickQWidget()

# Clear Layout - Begin
if afd2_window.layout() is not None:
	old_layout = afd2_window.layout()
	for i in reversed(range(old_layout.count())):
		old_layout.itemAt(i).widget().setParent(None)
	sip.delete(old_layout)
# Clear Layout - End

QtWidgets.QWidget.setLayout(afd2_window,afd2_qVboxLayout)
afd2_window.showFullScreen()
afd2_app.exec_()
